scripts-available/CDB_Quota.sql

```diff
@@ -3,50 +3,37 @@ CREATE OR REPLACE FUNCTION CDB_UserDataSize(schema_name TEXT)
 RETURNS bigint AS
 $$
 DECLARE
-  quota_vector INT8;
-  quota_raster INT8;
+  total_size INT8;
 BEGIN
-  -- TODO: double check queries. Maybe use CDB_TableMetadata for lookup?
-  --  Also, "table_name" sounds sensible to search_path
-
-  -- Division by 2 is for not counting the_geom_webmercator
-  SELECT COALESCE(INT8(SUM(pg_total_relation_size('"' || schema_name || '"."' || table_name || '"')) / 2), 0) INTO quota_vector
-  FROM information_schema.tables
-  WHERE table_catalog = current_database() AND table_schema = schema_name
-    AND table_name != 'spatial_ref_sys'
-    AND table_name != 'cdb_tablemetadata'
-    AND table_type = 'BASE TABLE'
-    -- exclude raster overview tables
-    AND table_name NOT IN (
-      SELECT o_table_name FROM raster_overviews
+  WITH raster_tables AS (
+    SELECT o_table_name, r_table_name FROM raster_overviews
       WHERE o_table_schema = schema_name AND o_table_catalog = current_database()
-    )
-    -- exclude raster "main" tables
-    AND table_name NOT IN (
-      SELECT r_table_name FROM raster_overviews
-      WHERE r_table_name = table_name
-        AND o_table_schema = schema_name AND o_table_catalog = current_database()
-    );
-
-  SELECT COALESCE(INT8(SUM(pg_total_relation_size('"' || schema_name || '"."' || table_name || '"'))), 0) INTO quota_raster
-  FROM information_schema.tables
-  WHERE table_catalog = current_database() AND table_schema = schema_name
-    AND table_name != 'spatial_ref_sys'
-    AND table_name != 'cdb_tablemetadata'
-    AND table_type = 'BASE TABLE'
-    -- exclude raster overview tables
-    AND table_name NOT IN (
-      SELECT o_table_name FROM raster_overviews
-      WHERE o_table_schema = schema_name AND o_table_catalog = current_database()
-    )
-    -- filter to raster "main" tables
-    AND table_name IN (
-      SELECT r_table_name FROM raster_overviews
-      WHERE r_table_name = table_name
-        AND o_table_schema = schema_name AND o_table_catalog = current_database()
-    );
-
-  RETURN quota_vector + quota_raster;
+  ),
+  user_tables AS (
+    SELECT table_name FROM information_schema.tables
+      WHERE table_catalog = current_database() AND table_schema = schema_name
+        AND table_name != 'spatial_ref_sys'
+        AND table_name != 'cdb_tablemetadata'
+        AND table_type = 'BASE TABLE'
+  ),
+  table_cat AS (
+    SELECT
+      table_name,
+      EXISTS(select * from raster_tables where o_table_name = table_name) AS is_overview,
```

Comment:
Ok, it's taken a while for me understood what's this. I'd suggest two minor improvements, please feel free to ignore them, since this version is a huge improvement from the previous one and I'm not familiarised with raster data schema:
* In 22-23, `select 1` instead of `select *` should work and db doesn't need to move any data (although planner could realise and do be this for us). 
* Since you're not accounting overview tables (you don't use tables that get `is_overview` true at all) you could try to avoid getting them in `raster_tables` so you don't need to check for existence in `table_cat` nor multiply it by 0 in `sizes`. I think picking only distinct r_table_name would work. Because of `case` precedence maybe you should also exclude the tables where `o_table_name = r_table_name` from raster_tables, but this might make no sense at all according to its meaning (I think the only actual use for `is_overview` is avoiding continue to `is_raster` or `else` cases).